HTTP 304 Not Modified Error

After scheduling Dependency Track Upload Step on Azure DevOps Services, I have got HTTP 304 Not Modified error.

##[error]Failed to update project: 304 - Not Modified


How can we handle this Not to give any error on Azure DevOps Services pipeline?

Thanks.

Hi @Zargath,

Here is the whole error log:

2025-03-17T02:10:10.5070162Z ##[section]Starting: Upload BOM to https://xx.endpoints.xxxx.cloud.goog
2025-03-17T02:10:10.5086243Z ==============================================================================
2025-03-17T02:10:10.5086391Z Task         : Upload BOM to Dependency Track
2025-03-17T02:10:10.5086464Z Description  : Upload a BOM file to Dependency Track platform.
2025-03-17T02:10:10.5086579Z Version      : 1.6.0
2025-03-17T02:10:10.5086638Z Author       : Edouard Shaar
2025-03-17T02:10:10.5086707Z Help         : 
2025-03-17T02:10:10.5086771Z ==============================================================================
2025-03-17T02:10:11.4398414Z Reading BOM from location: D:\a\63\a\_manifest\cyclone\bom.json
2025-03-17T02:10:11.4398911Z Starting BOM upload to Dependency Track for project with id 'fffffffff-ffff-ffff-ffff-ffffffffffff' at url: https://xx.endpoints.xxxx.cloud.goog
2025-03-17T02:10:11.4648378Z Uploading the BOM succeeded! Token: ffffffff-ffff-ffff-ffff-ffffffffffff
2025-03-17T02:10:11.4649178Z Updating project.
2025-03-17T02:10:11.6388417Z Error: <ref *2> IncomingMessage {
2025-03-17T02:10:11.6388703Z   _events: {
2025-03-17T02:10:11.6388921Z     close: [ [Function (anonymous)], [Function (anonymous)] ],
2025-03-17T02:10:11.6389140Z     error: [Function (anonymous)],
2025-03-17T02:10:11.6389336Z     data: [Function (anonymous)],
2025-03-17T02:10:11.6389529Z     end: [ [Function: responseOnEnd], [Function (anonymous)] ],
2025-03-17T02:10:11.6389731Z     readable: undefined
2025-03-17T02:10:11.6389901Z   },
2025-03-17T02:10:11.6390333Z   _readableState: ReadableState {
2025-03-17T02:10:11.6390518Z     highWaterMark: 16384,
2025-03-17T02:10:11.6390685Z     buffer: [],
2025-03-17T02:10:11.6390846Z     bufferIndex: 0,
2025-03-17T02:10:11.6390993Z     length: 0,
2025-03-17T02:10:11.6391149Z     pipes: [],
2025-03-17T02:10:11.6391332Z     awaitDrainWriters: null,
2025-03-17T02:10:11.6391500Z     [Symbol(kState)]: 194254604
2025-03-17T02:10:11.6391672Z   },
2025-03-17T02:10:11.6391848Z   _maxListeners: undefined,
2025-03-17T02:10:11.6392015Z   socket: null,
2025-03-17T02:10:11.6392173Z   httpVersionMajor: 1,
2025-03-17T02:10:11.6392342Z   httpVersionMinor: 1,
2025-03-17T02:10:11.6392518Z   httpVersion: '1.1',
2025-03-17T02:10:11.6392687Z   complete: true,
2025-03-17T02:10:11.6392836Z   rawHeaders: [
2025-03-17T02:10:11.6393005Z     'Date',
2025-03-17T02:10:11.6393205Z     'Mon, 17 Mar 2025 02:10:11 GMT',
2025-03-17T02:10:11.6393378Z     'X-Powered-By',
2025-03-17T02:10:11.6393566Z     'Dependency-Track v4.12.6',
2025-03-17T02:10:11.6393751Z     'Cache-Control',
2025-03-17T02:10:11.6393937Z     'private, max-age=0, must-revalidate, no-cache',
2025-03-17T02:10:11.6394130Z     'Access-Control-Allow-Origin',
2025-03-17T02:10:11.6394292Z     '*',
2025-03-17T02:10:11.6394457Z     'Access-Control-Allow-Methods',
2025-03-17T02:10:11.6394636Z     'GET, POST, PUT, DELETE, OPTIONS',
2025-03-17T02:10:11.6394825Z     'Access-Control-Allow-Headers',
2025-03-17T02:10:11.6395334Z     'Origin, Content-Type, Authorization, X-Requested-With, Content-Length, Accept, Origin, X-Api-Key, X-Total-Count, *',
2025-03-17T02:10:11.6395589Z     'Access-Control-Expose-Headers',
2025-03-17T02:10:11.6395838Z     'Origin, Content-Type, Authorization, X-Requested-With, Content-Length, Accept, Origin, X-Api-Key, X-Total-Count',
2025-03-17T02:10:11.6396097Z     'Access-Control-Allow-Credentials',
2025-03-17T02:10:11.6396279Z     'true',
2025-03-17T02:10:11.6396449Z     'Access-Control-Max-Age',
2025-03-17T02:10:11.6396603Z     '3600',
2025-03-17T02:10:11.6396773Z     'X-Request-Id',
2025-03-17T02:10:11.6396958Z     '35575628-af2c-4204-8f72-b6bc2f738d6e',
2025-03-17T02:10:11.6397151Z     'Content-Security-Policy',
2025-03-17T02:10:11.6397501Z     "default-src 'self';script-src 'self' 'unsafe-inline';style-src 'self' 'unsafe-inline';img-src 'self';connect-src 'self';font-src 'self';object-src 'self';media-src 'self';frame-src 'self';child-src 'self';form-action 'self';frame-ancestors 'none';",
2025-03-17T02:10:11.6399549Z     'X-Frame-Options',
2025-03-17T02:10:11.6399718Z     'DENY',
2025-03-17T02:10:11.6399886Z     'Content-Length',
2025-03-17T02:10:11.6400030Z     '0',
2025-03-17T02:10:11.6400198Z     'Via',
2025-03-17T02:10:11.6400358Z     '1.1 google',
2025-03-17T02:10:11.6400507Z     'Alt-Svc',
2025-03-17T02:10:11.6400706Z     'h3=":443"; ma=2592000,h3-29=":443"; ma=2592000'
2025-03-17T02:10:11.6400896Z   ],
2025-03-17T02:10:11.6401054Z   rawTrailers: [],
2025-03-17T02:10:11.6401219Z   joinDuplicateHeaders: undefined,
2025-03-17T02:10:11.6401402Z   aborted: false,
2025-03-17T02:10:11.6401576Z   upgrade: false,
2025-03-17T02:10:11.6401718Z   url: '',
2025-03-17T02:10:11.6401885Z   method: null,
2025-03-17T02:10:11.6402048Z   statusCode: 304,
2025-03-17T02:10:11.6402228Z   statusMessage: 'Not Modified',
2025-03-17T02:10:11.6402399Z   client: <ref *1> TLSSocket {
2025-03-17T02:10:11.6402574Z     _tlsOptions: {
2025-03-17T02:10:11.6402757Z       allowHalfOpen: undefined,
2025-03-17T02:10:11.6402929Z       pipe: false,
2025-03-17T02:10:11.6403092Z       secureContext: [SecureContext],
2025-03-17T02:10:11.6403270Z       isServer: false,
2025-03-17T02:10:11.6403437Z       requestCert: true,
2025-03-17T02:10:11.6403600Z       rejectUnauthorized: true,
2025-03-17T02:10:11.6403775Z       session: undefined,
2025-03-17T02:10:11.6403947Z       ALPNProtocols: undefined,
2025-03-17T02:10:11.6404124Z       requestOCSP: undefined,
2025-03-17T02:10:11.6404291Z       enableTrace: undefined,
2025-03-17T02:10:11.6404463Z       pskCallback: undefined,
2025-03-17T02:10:11.6404733Z       highWaterMark: undefined,
2025-03-17T02:10:11.6404900Z       onread: undefined,
2025-03-17T02:10:11.6405065Z       signal: undefined
2025-03-17T02:10:11.6405226Z     },
2025-03-17T02:10:11.6405387Z     _secureEstablished: true,
2025-03-17T02:10:11.6405555Z     _securePending: false,
2025-03-17T02:10:11.6405734Z     _newSessionPending: false,
2025-03-17T02:10:11.6405957Z     _controlReleased: true,
2025-03-17T02:10:11.6406124Z     secureConnecting: false,
2025-03-17T02:10:11.6406292Z     _SNICallback: null,
2025-03-17T02:10:11.6406488Z     servername: 'xx.endpoints.xxxx.cloud.goog',
2025-03-17T02:10:11.6406693Z     alpnProtocol: false,
2025-03-17T02:10:11.6406853Z     authorized: true,
2025-03-17T02:10:11.6407033Z     authorizationError: null,
2025-03-17T02:10:11.6407210Z     encrypted: true,
2025-03-17T02:10:11.6407396Z     _events: [Object: null prototype] {
2025-03-17T02:10:11.6407560Z       close: [Array],
2025-03-17T02:10:11.6407742Z       end: [Function: onReadableStreamEnd],
2025-03-17T02:10:11.6407919Z       error: undefined,
2025-03-17T02:10:11.6408095Z       newListener: [Function: keylogNewListener],
2025-03-17T02:10:11.6408284Z       connect: undefined,
2025-03-17T02:10:11.6408485Z       secure: [Function: onConnectSecure],
2025-03-17T02:10:11.6408675Z       session: [Function (anonymous)],
2025-03-17T02:10:11.6408853Z       free: [Function: onFree],
2025-03-17T02:10:11.6409054Z       timeout: [Function: onTimeout],
2025-03-17T02:10:11.6409260Z       agentRemove: [Function: onRemove],
2025-03-17T02:10:11.6409445Z       data: undefined,
2025-03-17T02:10:11.6409602Z       drain: undefined
2025-03-17T02:10:11.6409751Z     },
2025-03-17T02:10:11.6409905Z     _eventsCount: 8,
2025-03-17T02:10:11.6410060Z     connecting: false,
2025-03-17T02:10:11.6410231Z     _hadError: false,
2025-03-17T02:10:11.6410402Z     _parent: null,
2025-03-17T02:10:11.6410593Z     _host: 'xx.endpoints.xxxx.cloud.goog',
2025-03-17T02:10:11.6410785Z     _closeAfterHandlingError: false,
2025-03-17T02:10:11.6410974Z     _readableState: ReadableState {
2025-03-17T02:10:11.6411167Z       highWaterMark: 16384,
2025-03-17T02:10:11.6411321Z       buffer: [],
2025-03-17T02:10:11.6411492Z       bufferIndex: 0,
2025-03-17T02:10:11.6411670Z       length: 0,
2025-03-17T02:10:11.6411833Z       pipes: [],
2025-03-17T02:10:11.6411987Z       awaitDrainWriters: null,
2025-03-17T02:10:11.6412181Z       [Symbol(kState)]: 60303620
2025-03-17T02:10:11.6412749Z     },
2025-03-17T02:10:11.6412902Z     _writableState: WritableState {
2025-03-17T02:10:11.6413093Z       highWaterMark: 16384,
2025-03-17T02:10:11.6413261Z       length: 0,
2025-03-17T02:10:11.6413417Z       corked: 0,
2025-03-17T02:10:11.6413582Z       onwrite: [Function: bound onwrite],
2025-03-17T02:10:11.6413765Z       writelen: 0,
2025-03-17T02:10:11.6413941Z       bufferedIndex: 0,
2025-03-17T02:10:11.6414108Z       pendingcb: 0,
2025-03-17T02:10:11.6414267Z       [Symbol(kState)]: 17563908,
2025-03-17T02:10:11.6414456Z       [Symbol(kBufferedValue)]: null,
2025-03-17T02:10:11.6414641Z       [Symbol(kWriteCbValue)]: null
2025-03-17T02:10:11.6414789Z     },
2025-03-17T02:10:11.6414946Z     allowHalfOpen: false,
2025-03-17T02:10:11.6415125Z     _maxListeners: undefined,
2025-03-17T02:10:11.6415306Z     _sockname: null,
2025-03-17T02:10:11.6415461Z     _pendingData: null,
2025-03-17T02:10:11.6415632Z     _pendingEncoding: '',
2025-03-17T02:10:11.6415803Z     server: undefined,
2025-03-17T02:10:11.6415956Z     _server: null,
2025-03-17T02:10:11.6416121Z     ssl: TLSWrap {
2025-03-17T02:10:11.6416281Z       _parent: [TCP],
2025-03-17T02:10:11.6416447Z       _parentWrap: null,
2025-03-17T02:10:11.6416614Z       _secureContext: [SecureContext],
2025-03-17T02:10:11.6416787Z       reading: true,
2025-03-17T02:10:11.6416967Z       onkeylog: [Function: onkeylog],
2025-03-17T02:10:11.6417133Z       onhandshakestart: {},
2025-03-17T02:10:11.6417330Z       onhandshakedone: [Function (anonymous)],
2025-03-17T02:10:11.6417613Z       onocspresponse: [Function: onocspresponse],
2025-03-17T02:10:11.6417827Z       onnewsession: [Function: onnewsessionclient],
2025-03-17T02:10:11.6418016Z       onerror: [Function: onerror],
2025-03-17T02:10:11.6418202Z       [Symbol(owner_symbol)]: [Circular *1],
2025-03-17T02:10:11.6418414Z       [Symbol(resource_symbol)]: [ReusedHandle]
2025-03-17T02:10:11.6418592Z     },
2025-03-17T02:10:11.6418739Z     _requestCert: true,
2025-03-17T02:10:11.6418919Z     _rejectUnauthorized: true,
2025-03-17T02:10:11.6419092Z     timeout: 0,
2025-03-17T02:10:11.6419244Z     parser: null,
2025-03-17T02:10:11.6419420Z     _httpMessage: ClientRequest {
2025-03-17T02:10:11.6419605Z       _events: [Object: null prototype],
2025-03-17T02:10:11.6419785Z       _eventsCount: 5,
2025-03-17T02:10:11.6419947Z       _maxListeners: undefined,
2025-03-17T02:10:11.6420120Z       outputData: [],
2025-03-17T02:10:11.6420284Z       outputSize: 0,
2025-03-17T02:10:11.6420450Z       writable: true,
2025-03-17T02:10:11.6420602Z       destroyed: true,
2025-03-17T02:10:11.6420772Z       _last: false,
2025-03-17T02:10:11.6420938Z       chunkedEncoding: false,
2025-03-17T02:10:11.6421218Z       shouldKeepAlive: true,
2025-03-17T02:10:11.6421418Z       maxRequestsOnConnectionReached: false,
2025-03-17T02:10:11.6421612Z       _defaultKeepAlive: true,
2025-03-17T02:10:11.6421797Z       useChunkedEncodingByDefault: true,
2025-03-17T02:10:11.6421966Z       sendDate: false,
2025-03-17T02:10:11.6422143Z       _removedConnection: false,
2025-03-17T02:10:11.6422322Z       _removedContLen: false,
2025-03-17T02:10:11.6422483Z       _removedTE: false,
2025-03-17T02:10:11.6422654Z       strictContentLength: false,
2025-03-17T02:10:11.6422850Z       _contentLength: 28,
2025-03-17T02:10:11.6423024Z       _hasBody: true,
2025-03-17T02:10:11.6423174Z       _trailer: '',
2025-03-17T02:10:11.6423343Z       finished: true,
2025-03-17T02:10:11.6423506Z       _headerSent: true,
2025-03-17T02:10:11.6423662Z       _closed: false,
2025-03-17T02:10:11.6423825Z       socket: [Circular *1],
2025-03-17T02:10:11.6424045Z       _header: 'PATCH /api/v1/project/fffffffff-ffff-ffff-ffff-ffffffffffff HTTP/1.1\r\n' +
2025-03-17T02:10:11.6424289Z         'X-API-Key: odt_ffffffffffffffffffffffffffffffff\r\n' +
2025-03-17T02:10:11.6424501Z         'host: xx.endpoints.xxxx.cloud.goog\r\n' +
2025-03-17T02:10:11.6424700Z         'accept: application/json\r\n' +
2025-03-17T02:10:11.6424958Z         'content-type: application/json\r\n' +
2025-03-17T02:10:11.6425143Z         'content-length: 28\r\n' +
2025-03-17T02:10:11.6425319Z         'Connection: keep-alive\r\n' +
2025-03-17T02:10:11.6425484Z         '\r\n',
2025-03-17T02:10:11.6425657Z       _keepAliveTimeout: 0,
2025-03-17T02:10:11.6425838Z       _onPendingData: [Function: nop],
2025-03-17T02:10:11.6426002Z       agent: [Agent],
2025-03-17T02:10:11.6426321Z       socketPath: undefined,
2025-03-17T02:10:11.6426550Z       method: 'PATCH',
2025-03-17T02:10:11.6426716Z       maxHeaderSize: undefined,
2025-03-17T02:10:11.6426909Z       insecureHTTPParser: undefined,
2025-03-17T02:10:11.6427105Z       joinDuplicateHeaders: undefined,
2025-03-17T02:10:11.6427326Z       path: '/api/v1/project/fffffffff-ffff-ffff-ffff-ffffffffffff',
2025-03-17T02:10:11.6427517Z       _ended: true,
2025-03-17T02:10:11.6427684Z       res: [Circular *2],
2025-03-17T02:10:11.6427858Z       aborted: false,
2025-03-17T02:10:11.6428033Z       timeoutCb: null,
2025-03-17T02:10:11.6428195Z       upgradeOrConnect: false,
2025-03-17T02:10:11.6428366Z       parser: null,
2025-03-17T02:10:11.6428534Z       maxHeadersCount: null,
2025-03-17T02:10:11.6428696Z       reusedSocket: true,
2025-03-17T02:10:11.6428885Z       host: 'xx.endpoints.xxxx.cloud.goog',
2025-03-17T02:10:11.6429090Z       protocol: 'https:',
2025-03-17T02:10:11.6429267Z       [Symbol(shapeMode)]: false,
2025-03-17T02:10:11.6429435Z       [Symbol(kCapture)]: false,
2025-03-17T02:10:11.6429702Z       [Symbol(kBytesWritten)]: 0,
2025-03-17T02:10:11.6429882Z       [Symbol(kNeedDrain)]: false,
2025-03-17T02:10:11.6430044Z       [Symbol(corked)]: 0,
2025-03-17T02:10:11.6430234Z       [Symbol(kOutHeaders)]: [Object: null prototype],
2025-03-17T02:10:11.6430425Z       [Symbol(errored)]: null,
2025-03-17T02:10:11.6430614Z       [Symbol(kHighWaterMark)]: 16384,
2025-03-17T02:10:11.6430800Z       [Symbol(kRejectNonStandardBodyWrites)]: false,
2025-03-17T02:10:11.6431004Z       [Symbol(kUniqueHeaders)]: null
2025-03-17T02:10:11.6431170Z     },
2025-03-17T02:10:11.6431335Z     [Symbol(alpncallback)]: null,
2025-03-17T02:10:11.6431659Z     [Symbol(res)]: TLSWrap {
2025-03-17T02:10:11.6431828Z       _parent: [TCP],
2025-03-17T02:10:11.6431997Z       _parentWrap: null,
2025-03-17T02:10:11.6432164Z       _secureContext: [SecureContext],
2025-03-17T02:10:11.6432343Z       reading: true,
2025-03-17T02:10:11.6432516Z       onkeylog: [Function: onkeylog],
2025-03-17T02:10:11.6432705Z       onhandshakestart: {},
2025-03-17T02:10:11.6432885Z       onhandshakedone: [Function (anonymous)],
2025-03-17T02:10:11.6433089Z       onocspresponse: [Function: onocspresponse],
2025-03-17T02:10:11.6433303Z       onnewsession: [Function: onnewsessionclient],
2025-03-17T02:10:11.6433503Z       onerror: [Function: onerror],
2025-03-17T02:10:11.6433679Z       [Symbol(owner_symbol)]: [Circular *1],
2025-03-17T02:10:11.6433876Z       [Symbol(resource_symbol)]: [ReusedHandle]
2025-03-17T02:10:11.6434062Z     },
2025-03-17T02:10:11.6434211Z     [Symbol(verified)]: true,
2025-03-17T02:10:11.6434391Z     [Symbol(pendingSession)]: null,
2025-03-17T02:10:11.6434590Z     [Symbol(async_id_symbol)]: 56,
2025-03-17T02:10:11.6434773Z     [Symbol(kHandle)]: TLSWrap {
2025-03-17T02:10:11.6434934Z       _parent: [TCP],
2025-03-17T02:10:11.6435096Z       _parentWrap: null,
2025-03-17T02:10:11.6435281Z       _secureContext: [SecureContext],
2025-03-17T02:10:11.6435453Z       reading: true,
2025-03-17T02:10:11.6435616Z       onkeylog: [Function: onkeylog],
2025-03-17T02:10:11.6435794Z       onhandshakestart: {},
2025-03-17T02:10:11.6435987Z       onhandshakedone: [Function (anonymous)],
2025-03-17T02:10:11.6436181Z       onocspresponse: [Function: onocspresponse],
2025-03-17T02:10:11.6436393Z       onnewsession: [Function: onnewsessionclient],
2025-03-17T02:10:11.6436744Z       onerror: [Function: onerror],
2025-03-17T02:10:11.6436942Z       [Symbol(owner_symbol)]: [Circular *1],
2025-03-17T02:10:11.6437196Z       [Symbol(resource_symbol)]: [ReusedHandle]
2025-03-17T02:10:11.6437364Z     },
2025-03-17T02:10:11.6437540Z     [Symbol(lastWriteQueueSize)]: 0,
2025-03-17T02:10:11.6437737Z     [Symbol(timeout)]: Timeout {
2025-03-17T02:10:11.6437903Z       _idleTimeout: -1,
2025-03-17T02:10:11.6438074Z       _idlePrev: null,
2025-03-17T02:10:11.6438237Z       _idleNext: null,
2025-03-17T02:10:11.6438388Z       _idleStart: 982,
2025-03-17T02:10:11.6438556Z       _onTimeout: null,
2025-03-17T02:10:11.6438734Z       _timerArgs: undefined,
2025-03-17T02:10:11.6438903Z       _repeat: null,
2025-03-17T02:10:11.6439055Z       _destroyed: true,
2025-03-17T02:10:11.6439232Z       [Symbol(refed)]: false,
2025-03-17T02:10:11.6439414Z       [Symbol(kHasPrimitive)]: false,
2025-03-17T02:10:11.6439580Z       [Symbol(asyncId)]: 51,
2025-03-17T02:10:11.6439755Z       [Symbol(triggerId)]: 49
2025-03-17T02:10:11.6439905Z     },
2025-03-17T02:10:11.6440079Z     [Symbol(kBuffer)]: null,
2025-03-17T02:10:11.6440245Z     [Symbol(kBufferCb)]: null,
2025-03-17T02:10:11.6440426Z     [Symbol(kBufferGen)]: null,
2025-03-17T02:10:11.6440602Z     [Symbol(shapeMode)]: true,
2025-03-17T02:10:11.6440768Z     [Symbol(kCapture)]: false,
2025-03-17T02:10:11.6440945Z     [Symbol(kSetNoDelay)]: false,
2025-03-17T02:10:11.6441127Z     [Symbol(kSetKeepAlive)]: true,
2025-03-17T02:10:11.6441320Z     [Symbol(kSetKeepAliveInitialDelay)]: 1,
2025-03-17T02:10:11.6441497Z     [Symbol(kBytesRead)]: 0,
2025-03-17T02:10:11.6441683Z     [Symbol(kBytesWritten)]: 0,
2025-03-17T02:10:11.6442089Z     [Symbol(connect-options)]: {
2025-03-17T02:10:11.6442272Z       rejectUnauthorized: true,
2025-03-17T02:10:11.6442889Z       ciphers: 'TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256:TLS_AES_128_GCM_SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA256:ECDHE-RSA-AES256-SHA384:DHE-RSA-AES256-SHA384:ECDHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA256:HIGH:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!PSK:!SRP:!CAMELLIA',
2025-03-17T02:10:11.6443540Z       checkServerIdentity: [Function: checkServerIdentity],
2025-03-17T02:10:11.6443758Z       minDHSize: 1024,
2025-03-17T02:10:11.6443948Z       _events: [Object: null prototype],
2025-03-17T02:10:11.6444113Z       _eventsCount: 5,
2025-03-17T02:10:11.6444293Z       _maxListeners: undefined,
2025-03-17T02:10:11.6444467Z       headers: [Object],
2025-03-17T02:10:11.6444637Z       method: 'POST',
2025-03-17T02:10:11.6444795Z       formData: [Object],
2025-03-17T02:10:11.6444965Z       uri: [Url],
2025-03-17T02:10:11.6445148Z       callback: [Function (anonymous)],
2025-03-17T02:10:11.6445331Z       readable: true,
2025-03-17T02:10:11.6445483Z       writable: true,
2025-03-17T02:10:11.6445657Z       explicitMethod: true,
2025-03-17T02:10:11.6445847Z       _qs: [Querystring],
2025-03-17T02:10:11.6446000Z       _auth: [Auth],
2025-03-17T02:10:11.6446181Z       _oauth: [OAuth],
2025-03-17T02:10:11.6446353Z       _multipart: [Multipart],
2025-03-17T02:10:11.6446534Z       _redirect: [Redirect],
2025-03-17T02:10:11.6446695Z       _tunnel: [Tunnel],
2025-03-17T02:10:11.6446877Z       setHeader: [Function (anonymous)],
2025-03-17T02:10:11.6447257Z       hasHeader: [Function (anonymous)],
2025-03-17T02:10:11.6447436Z       getHeader: [Function (anonymous)],
2025-03-17T02:10:11.6447631Z       removeHeader: [Function (anonymous)],
2025-03-17T02:10:11.6447826Z       localAddress: undefined,
2025-03-17T02:10:11.6447995Z       pool: {},
2025-03-17T02:10:11.6448139Z       dests: [],
2025-03-17T02:10:11.6448314Z       __isRequestRequest: true,
2025-03-17T02:10:11.6448513Z       _callback: [Function (anonymous)],
2025-03-17T02:10:11.6448687Z       proxy: null,
2025-03-17T02:10:11.6448837Z       tunnel: true,
2025-03-17T02:10:11.6449007Z       setHost: true,
2025-03-17T02:10:11.6449180Z       originalCookieHeader: undefined,
2025-03-17T02:10:11.6449841Z       _disableCookies: true,
2025-03-17T02:10:11.6450030Z       _jar: undefined,
2025-03-17T02:10:11.6450194Z       port: 443,
2025-03-17T02:10:11.6450383Z       host: 'xx.endpoints.xxxx.cloud.goog',
2025-03-17T02:10:11.6450561Z       _form: [FormData],
2025-03-17T02:10:11.6450721Z       path: null,
2025-03-17T02:10:11.6450894Z       _json: true,
2025-03-17T02:10:11.6451051Z       httpModule: [Object],
2025-03-17T02:10:11.6451241Z       agentClass: [Function: Agent],
2025-03-17T02:10:11.6451424Z       agent: [Agent],
2025-03-17T02:10:11.6451584Z       ntick: true,
2025-03-17T02:10:11.6451735Z       src: [FormData],
2025-03-17T02:10:11.6451900Z       _started: true,
2025-03-17T02:10:11.6452217Z       href: 'https://xx.endpoints.xxxx.cloud.goog/api/v1/bom',
2025-03-17T02:10:11.6452463Z       _defaultAgent: [Agent],
2025-03-17T02:10:11.6452648Z       keepAlive: true,
2025-03-17T02:10:11.6452828Z       scheduling: 'lifo',
2025-03-17T02:10:11.6452994Z       timeout: 5000,
2025-03-17T02:10:11.6453146Z       noDelay: true,
2025-03-17T02:10:11.6453347Z       servername: 'xx.endpoints.xxxx.cloud.goog',
2025-03-17T02:10:11.6453608Z       _agentKey: 'xx.endpoints.xxxx.cloud.goog:443:::::::::::::::::::::',
2025-03-17T02:10:11.6453833Z       encoding: null,
2025-03-17T02:10:11.6453994Z       keepAliveInitialDelay: 1000
2025-03-17T02:10:11.6454155Z     }
2025-03-17T02:10:11.6454292Z   },
2025-03-17T02:10:11.6454435Z   _consuming: false,
2025-03-17T02:10:11.6454635Z   _dumped: false,
2025-03-17T02:10:11.6454886Z   req: <ref *3> ClientRequest {
2025-03-17T02:10:11.6455080Z     _events: [Object: null prototype] {
2025-03-17T02:10:11.6455253Z       response: [Function: bound ],
2025-03-17T02:10:11.6455442Z       error: [Function: bound ],
2025-03-17T02:10:11.6455624Z       drain: [Function (anonymous)],
2025-03-17T02:10:11.6455812Z       socket: [Function (anonymous)],
2025-03-17T02:10:11.6455989Z       finish: [Function: requestOnFinish]
2025-03-17T02:10:11.6456173Z     },
2025-03-17T02:10:11.6456336Z     _eventsCount: 5,
2025-03-17T02:10:11.6456495Z     _maxListeners: undefined,
2025-03-17T02:10:11.6456670Z     outputData: [],
2025-03-17T02:10:11.6456843Z     outputSize: 0,
2025-03-17T02:10:11.6457020Z     writable: true,
2025-03-17T02:10:11.6457172Z     destroyed: true,
2025-03-17T02:10:11.6457500Z     _last: false,
2025-03-17T02:10:11.6457685Z     chunkedEncoding: false,
2025-03-17T02:10:11.6457852Z     shouldKeepAlive: true,
2025-03-17T02:10:11.6458040Z     maxRequestsOnConnectionReached: false,
2025-03-17T02:10:11.6458233Z     _defaultKeepAlive: true,
2025-03-17T02:10:11.6458424Z     useChunkedEncodingByDefault: true,
2025-03-17T02:10:11.6458590Z     sendDate: false,
2025-03-17T02:10:11.6458777Z     _removedConnection: false,
2025-03-17T02:10:11.6458966Z     _removedContLen: false,
2025-03-17T02:10:11.6459127Z     _removedTE: false,
2025-03-17T02:10:11.6459302Z     strictContentLength: false,
2025-03-17T02:10:11.6459487Z     _contentLength: 28,
2025-03-17T02:10:11.6459665Z     _hasBody: true,
2025-03-17T02:10:11.6459816Z     _trailer: '',
2025-03-17T02:10:11.6459975Z     finished: true,
2025-03-17T02:10:11.6460146Z     _headerSent: true,
2025-03-17T02:10:11.6460299Z     _closed: false,
2025-03-17T02:10:11.6460474Z     socket: <ref *1> TLSSocket {
2025-03-17T02:10:11.6460654Z       _tlsOptions: [Object],
2025-03-17T02:10:11.6460841Z       _secureEstablished: true,
2025-03-17T02:10:11.6461007Z       _securePending: false,
2025-03-17T02:10:11.6461195Z       _newSessionPending: false,
2025-03-17T02:10:11.6461398Z       _controlReleased: true,
2025-03-17T02:10:11.6461584Z       secureConnecting: false,
2025-03-17T02:10:11.6461747Z       _SNICallback: null,
2025-03-17T02:10:11.6461942Z       servername: 'xx.endpoints.xxxx.cloud.goog',
2025-03-17T02:10:11.6462135Z       alpnProtocol: false,
2025-03-17T02:10:11.6462291Z       authorized: true,
2025-03-17T02:10:11.6462579Z       authorizationError: null,
2025-03-17T02:10:11.6463332Z       encrypted: true,
2025-03-17T02:10:11.6463521Z       _events: [Object: null prototype],
2025-03-17T02:10:11.6463686Z       _eventsCount: 8,
2025-03-17T02:10:11.6463858Z       connecting: false,
2025-03-17T02:10:11.6464037Z       _hadError: false,
2025-03-17T02:10:11.6464188Z       _parent: null,
2025-03-17T02:10:11.6464393Z       _host: 'xx.endpoints.xxxx.cloud.goog',
2025-03-17T02:10:11.6464593Z       _closeAfterHandlingError: false,
2025-03-17T02:10:11.6464790Z       _readableState: [ReadableState],
2025-03-17T02:10:11.6464967Z       _writableState: [WritableState],
2025-03-17T02:10:11.6465154Z       allowHalfOpen: false,
2025-03-17T02:10:11.6465338Z       _maxListeners: undefined,
2025-03-17T02:10:11.6465506Z       _sockname: null,
2025-03-17T02:10:11.6465662Z       _pendingData: null,
2025-03-17T02:10:11.6465838Z       _pendingEncoding: '',
2025-03-17T02:10:11.6466008Z       server: undefined,
2025-03-17T02:10:11.6466162Z       _server: null,
2025-03-17T02:10:11.6466330Z       ssl: [TLSWrap],
2025-03-17T02:10:11.6466499Z       _requestCert: true,
2025-03-17T02:10:11.6466670Z       _rejectUnauthorized: true,
2025-03-17T02:10:11.6466826Z       timeout: 0,
2025-03-17T02:10:11.6466989Z       parser: null,
2025-03-17T02:10:11.6467156Z       _httpMessage: [Circular *3],
2025-03-17T02:10:11.6467327Z       [Symbol(alpncallback)]: null,
2025-03-17T02:10:11.6467514Z       [Symbol(res)]: [TLSWrap],
2025-03-17T02:10:11.6467690Z       [Symbol(verified)]: true,
2025-03-17T02:10:11.6468062Z       [Symbol(pendingSession)]: null,
2025-03-17T02:10:11.6468318Z       [Symbol(async_id_symbol)]: 56,
2025-03-17T02:10:11.6468499Z       [Symbol(kHandle)]: [TLSWrap],
2025-03-17T02:10:11.6468682Z       [Symbol(lastWriteQueueSize)]: 0,
2025-03-17T02:10:11.6468865Z       [Symbol(timeout)]: Timeout {
2025-03-17T02:10:11.6469028Z         _idleTimeout: -1,
2025-03-17T02:10:11.6469193Z         _idlePrev: null,
2025-03-17T02:10:11.6469364Z         _idleNext: null,
2025-03-17T02:10:11.6469515Z         _idleStart: 982,
2025-03-17T02:10:11.6469694Z         _onTimeout: null,
2025-03-17T02:10:11.6469862Z         _timerArgs: undefined,
2025-03-17T02:10:11.6470029Z         _repeat: null,
2025-03-17T02:10:11.6470184Z         _destroyed: true,
2025-03-17T02:10:11.6470352Z         [Symbol(refed)]: false,
2025-03-17T02:10:11.6470548Z         [Symbol(kHasPrimitive)]: false,
2025-03-17T02:10:11.6470726Z         [Symbol(asyncId)]: 51,
2025-03-17T02:10:11.6470888Z         [Symbol(triggerId)]: 49
2025-03-17T02:10:11.6471044Z       },
2025-03-17T02:10:11.6471208Z       [Symbol(kBuffer)]: null,
2025-03-17T02:10:11.6471373Z       [Symbol(kBufferCb)]: null,
2025-03-17T02:10:11.6471549Z       [Symbol(kBufferGen)]: null,
2025-03-17T02:10:11.6471736Z       [Symbol(shapeMode)]: true,
2025-03-17T02:10:11.6471916Z       [Symbol(kCapture)]: false,
2025-03-17T02:10:11.6472084Z       [Symbol(kSetNoDelay)]: false,
2025-03-17T02:10:11.6472283Z       [Symbol(kSetKeepAlive)]: true,
2025-03-17T02:10:11.6472486Z       [Symbol(kSetKeepAliveInitialDelay)]: 1,
2025-03-17T02:10:11.6472666Z       [Symbol(kBytesRead)]: 0,
2025-03-17T02:10:11.6472842Z       [Symbol(kBytesWritten)]: 0,
2025-03-17T02:10:11.6473182Z       [Symbol(connect-options)]: [Object]
2025-03-17T02:10:11.6473366Z     },
2025-03-17T02:10:11.6473561Z     _header: 'PATCH /api/v1/project/fffffffff-ffff-ffff-ffff-ffffffffffff HTTP/1.1\r\n' +
2025-03-17T02:10:11.6473818Z       'X-API-Key: odt_ffffffffffffffffffffffffffffffff\r\n' +
2025-03-17T02:10:11.6474040Z       'host: xx.endpoints.xxxx.cloud.goog\r\n' +
2025-03-17T02:10:11.6474241Z       'accept: application/json\r\n' +
2025-03-17T02:10:11.6474421Z       'content-type: application/json\r\n' +
2025-03-17T02:10:11.6474613Z       'content-length: 28\r\n' +
2025-03-17T02:10:11.6474792Z       'Connection: keep-alive\r\n' +
2025-03-17T02:10:11.6474955Z       '\r\n',
2025-03-17T02:10:11.6475107Z     _keepAliveTimeout: 0,
2025-03-17T02:10:11.6475299Z     _onPendingData: [Function: nop],
2025-03-17T02:10:11.6475570Z     agent: Agent {
2025-03-17T02:10:11.6475733Z       _events: [Object: null prototype],
2025-03-17T02:10:11.6475909Z       _eventsCount: 2,
2025-03-17T02:10:11.6476099Z       _maxListeners: undefined,
2025-03-17T02:10:11.6476275Z       defaultPort: 443,
2025-03-17T02:10:11.6476441Z       protocol: 'https:',
2025-03-17T02:10:11.6476629Z       options: [Object: null prototype],
2025-03-17T02:10:11.6476821Z       requests: [Object: null prototype] {},
2025-03-17T02:10:11.6476999Z       sockets: [Object: null prototype],
2025-03-17T02:10:11.6477216Z       freeSockets: [Object: null prototype] {},
2025-03-17T02:10:11.6477406Z       keepAliveMsecs: 1000,
2025-03-17T02:10:11.6477576Z       keepAlive: true,
2025-03-17T02:10:11.6477731Z       maxSockets: Infinity,
2025-03-17T02:10:11.6477909Z       maxFreeSockets: 256,
2025-03-17T02:10:11.6478075Z       scheduling: 'lifo',
2025-03-17T02:10:11.6478418Z       maxTotalSockets: Infinity,
2025-03-17T02:10:11.6478583Z       totalSocketCount: 1,
2025-03-17T02:10:11.6478765Z       maxCachedSessions: 100,
2025-03-17T02:10:11.6478946Z       _sessionCache: [Object],
2025-03-17T02:10:11.6479112Z       [Symbol(shapeMode)]: false,
2025-03-17T02:10:11.6479297Z       [Symbol(kCapture)]: false
2025-03-17T02:10:11.6479454Z     },
2025-03-17T02:10:11.6479626Z     socketPath: undefined,
2025-03-17T02:10:11.6479784Z     method: 'PATCH',
2025-03-17T02:10:11.6479966Z     maxHeaderSize: undefined,
2025-03-17T02:10:11.6480164Z     insecureHTTPParser: undefined,
2025-03-17T02:10:11.6480832Z     joinDuplicateHeaders: undefined,
2025-03-17T02:10:11.6481050Z     path: '/api/v1/project/fffffffff-ffff-ffff-ffff-ffffffffffff',
2025-03-17T02:10:11.6481266Z     _ended: true,
2025-03-17T02:10:11.6481435Z     res: [Circular *2],
2025-03-17T02:10:11.6481590Z     aborted: false,
2025-03-17T02:10:11.6481769Z     timeoutCb: null,
2025-03-17T02:10:11.6481962Z     upgradeOrConnect: false,
2025-03-17T02:10:11.6482131Z     parser: null,
2025-03-17T02:10:11.6482297Z     maxHeadersCount: null,
2025-03-17T02:10:11.6482468Z     reusedSocket: true,
2025-03-17T02:10:11.6482674Z     host: 'xx.endpoints.xxxx.cloud.goog',
2025-03-17T02:10:11.6482861Z     protocol: 'https:',
2025-03-17T02:10:11.6483050Z     [Symbol(shapeMode)]: false,
2025-03-17T02:10:11.6483231Z     [Symbol(kCapture)]: false,
2025-03-17T02:10:11.6483503Z     [Symbol(kBytesWritten)]: 0,
2025-03-17T02:10:11.6483673Z     [Symbol(kNeedDrain)]: false,
2025-03-17T02:10:11.6483854Z     [Symbol(corked)]: 0,
2025-03-17T02:10:11.6484046Z     [Symbol(kOutHeaders)]: [Object: null prototype] {
2025-03-17T02:10:11.6484229Z       'x-api-key': [Array],
2025-03-17T02:10:11.6484406Z       host: [Array],
2025-03-17T02:10:11.6484575Z       accept: [Array],
2025-03-17T02:10:11.6484743Z       'content-type': [Array],
2025-03-17T02:10:11.6484908Z       'content-length': [Array]
2025-03-17T02:10:11.6485067Z     },
2025-03-17T02:10:11.6485228Z     [Symbol(errored)]: null,
2025-03-17T02:10:11.6485411Z     [Symbol(kHighWaterMark)]: 16384,
2025-03-17T02:10:11.6485602Z     [Symbol(kRejectNonStandardBodyWrites)]: false,
2025-03-17T02:10:11.6485813Z     [Symbol(kUniqueHeaders)]: null
2025-03-17T02:10:11.6485978Z   },
2025-03-17T02:10:11.6486123Z   _eventsCount: 4,
2025-03-17T02:10:11.6486295Z   request: <ref *4> Request {
2025-03-17T02:10:11.6486496Z     _events: [Object: null prototype] {
2025-03-17T02:10:11.6486676Z       error: [Function: bound ],
2025-03-17T02:10:11.6486846Z       complete: [Function: bound ],
2025-03-17T02:10:11.6487026Z       pipe: [Function (anonymous)],
2025-03-17T02:10:11.6487207Z       data: [Function (anonymous)],
2025-03-17T02:10:11.6487375Z       end: [Function (anonymous)]
2025-03-17T02:10:11.6487538Z     },
2025-03-17T02:10:11.6487698Z     _eventsCount: 5,
2025-03-17T02:10:11.6487872Z     _maxListeners: undefined,
2025-03-17T02:10:11.6488025Z     headers: {
2025-03-17T02:10:11.6488213Z       'X-API-Key': 'odt_ffffffffffffffffffffffffffffffff',
2025-03-17T02:10:11.6488414Z       accept: 'application/json',
2025-03-17T02:10:11.6488695Z       'content-type': 'application/json',
2025-03-17T02:10:11.6488869Z       'content-length': 28
2025-03-17T02:10:11.6489028Z     },
2025-03-17T02:10:11.6489188Z     method: 'PATCH',
2025-03-17T02:10:11.6489333Z     uri: Url {
2025-03-17T02:10:11.6489908Z       protocol: 'https:',
2025-03-17T02:10:11.6490091Z       slashes: true,
2025-03-17T02:10:11.6490259Z       auth: null,
2025-03-17T02:10:11.6490430Z       host: 'xx.endpoints.xxxx.cloud.goog',
2025-03-17T02:10:11.6490617Z       port: 443,
2025-03-17T02:10:11.6490819Z       hostname: 'xx.endpoints.xxxx.cloud.goog',
2025-03-17T02:10:11.6490995Z       hash: null,
2025-03-17T02:10:11.6491162Z       search: null,
2025-03-17T02:10:11.6491328Z       query: null,
2025-03-17T02:10:11.6491524Z       pathname: '/api/v1/project/fffffffff-ffff-ffff-ffff-ffffffffffff',
2025-03-17T02:10:11.6491752Z       path: '/api/v1/project/fffffffff-ffff-ffff-ffff-ffffffffffff',
2025-03-17T02:10:11.6492019Z       href: 'https://xx.endpoints.xxxx.cloud.goog/api/v1/project/fffffffff-ffff-ffff-ffff-ffffffffffff'
2025-03-17T02:10:11.6492333Z     },
2025-03-17T02:10:11.6492515Z     callback: [Function (anonymous)],
2025-03-17T02:10:11.6492680Z     readable: true,
2025-03-17T02:10:11.6492906Z     writable: true,
2025-03-17T02:10:11.6493079Z     explicitMethod: true,
2025-03-17T02:10:11.6493239Z     _qs: Querystring {
2025-03-17T02:10:11.6493407Z       request: [Circular *4],
2025-03-17T02:10:11.6493569Z       lib: [Object],
2025-03-17T02:10:11.6493828Z       useQuerystring: undefined,
2025-03-17T02:10:11.6493994Z       parseOptions: {},
2025-03-17T02:10:11.6494170Z       stringifyOptions: {}
2025-03-17T02:10:11.6494333Z     },
2025-03-17T02:10:11.6494482Z     _auth: Auth {
2025-03-17T02:10:11.6494710Z       request: [Circular *4],
2025-03-17T02:10:11.6494892Z       hasAuth: false,
2025-03-17T02:10:11.6495056Z       sentAuth: false,
2025-03-17T02:10:11.6495213Z       bearerToken: null,
2025-03-17T02:10:11.6495377Z       user: null,
2025-03-17T02:10:11.6495548Z       pass: null
2025-03-17T02:10:11.6495694Z     },
2025-03-17T02:10:11.6495862Z     _oauth: OAuth { request: [Circular *4], params: null },
2025-03-17T02:10:11.6496069Z     _multipart: Multipart {
2025-03-17T02:10:11.6496244Z       request: [Circular *4],
2025-03-17T02:10:11.6496427Z       boundary: '54aafafc-fed9-4208-a5c8-364b430c1a58',
2025-03-17T02:10:11.6496620Z       chunked: false,
2025-03-17T02:10:11.6496784Z       body: null
2025-03-17T02:10:11.6496941Z     },
2025-03-17T02:10:11.6497088Z     _redirect: Redirect {
2025-03-17T02:10:11.6497269Z       request: [Circular *4],
2025-03-17T02:10:11.6497438Z       followRedirect: true,
2025-03-17T02:10:11.6497600Z       followRedirects: true,
2025-03-17T02:10:11.6497776Z       followAllRedirects: false,
2025-03-17T02:10:11.6497959Z       followOriginalHttpMethod: false,
2025-03-17T02:10:11.6498162Z       allowRedirect: [Function (anonymous)],
2025-03-17T02:10:11.6498336Z       maxRedirects: 10,
2025-03-17T02:10:11.6498501Z       redirects: [],
2025-03-17T02:10:11.6498679Z       redirectsFollowed: 0,
2025-03-17T02:10:11.6498857Z       removeRefererHeader: false
2025-03-17T02:10:11.6499004Z     },
2025-03-17T02:10:11.6499162Z     _tunnel: Tunnel {
2025-03-17T02:10:11.6499335Z       request: [Circular *4],
2025-03-17T02:10:11.6499502Z       proxyHeaderWhiteList: [Array],
2025-03-17T02:10:11.6499859Z       proxyHeaderExclusiveList: []
2025-03-17T02:10:11.6500020Z     },
2025-03-17T02:10:11.6500199Z     setHeader: [Function (anonymous)],
2025-03-17T02:10:11.6500375Z     hasHeader: [Function (anonymous)],
2025-03-17T02:10:11.6500562Z     getHeader: [Function (anonymous)],
2025-03-17T02:10:11.6500751Z     removeHeader: [Function (anonymous)],
2025-03-17T02:10:11.6500928Z     localAddress: undefined,
2025-03-17T02:10:11.6501107Z     pool: {},
2025-03-17T02:10:11.6501266Z     dests: [],
2025-03-17T02:10:11.6501441Z     __isRequestRequest: true,
2025-03-17T02:10:11.6501696Z     _callback: [Function (anonymous)],
2025-03-17T02:10:11.6501866Z     proxy: null,
2025-03-17T02:10:11.6502031Z     tunnel: true,
2025-03-17T02:10:11.6502180Z     setHost: true,
2025-03-17T02:10:11.6502360Z     originalCookieHeader: undefined,
2025-03-17T02:10:11.6502558Z     _disableCookies: true,
2025-03-17T02:10:11.6502735Z     _jar: undefined,
2025-03-17T02:10:11.6502882Z     port: 443,
2025-03-17T02:10:11.6503109Z     host: 'xx.endpoints.xxxx.cloud.goog',
2025-03-17T02:10:11.6503336Z     path: '/api/v1/project/fffffffff-ffff-ffff-ffff-ffffffffffff',
2025-03-17T02:10:11.6503542Z     _json: true,
2025-03-17T02:10:11.6503707Z     body: '{"classifier":"APPLICATION"}',
2025-03-17T02:10:11.6503894Z     httpModule: {
2025-03-17T02:10:11.6504071Z       Agent: [Function: Agent],
2025-03-17T02:10:11.6504236Z       globalAgent: [Agent],
2025-03-17T02:10:11.6504412Z       Server: [Function: Server],
2025-03-17T02:10:11.6504602Z       createServer: [Function: createServer],
2025-03-17T02:10:11.6504906Z       get: [Function: get],
2025-03-17T02:10:11.6505070Z       request: [Function: request]
2025-03-17T02:10:11.6505229Z     },
2025-03-17T02:10:11.6505399Z     agentClass: [Function: Agent],
2025-03-17T02:10:11.6505580Z     agent: Agent {
2025-03-17T02:10:11.6505744Z       _events: [Object: null prototype],
2025-03-17T02:10:11.6505926Z       _eventsCount: 2,
2025-03-17T02:10:11.6506101Z       _maxListeners: undefined,
2025-03-17T02:10:11.6506264Z       defaultPort: 443,
2025-03-17T02:10:11.6506445Z       protocol: 'https:',
2025-03-17T02:10:11.6506698Z       options: [Object: null prototype],
2025-03-17T02:10:11.6506904Z       requests: [Object: null prototype] {},
2025-03-17T02:10:11.6507082Z       sockets: [Object: null prototype],
2025-03-17T02:10:11.6507275Z       freeSockets: [Object: null prototype] {},
2025-03-17T02:10:11.6507465Z       keepAliveMsecs: 1000,
2025-03-17T02:10:11.6507621Z       keepAlive: true,
2025-03-17T02:10:11.6507788Z       maxSockets: Infinity,
2025-03-17T02:10:11.6507962Z       maxFreeSockets: 256,
2025-03-17T02:10:11.6508143Z       scheduling: 'lifo',
2025-03-17T02:10:11.6508308Z       maxTotalSockets: Infinity,
2025-03-17T02:10:11.6508501Z       totalSocketCount: 1,
2025-03-17T02:10:11.6508688Z       maxCachedSessions: 100,
2025-03-17T02:10:11.6508872Z       _sessionCache: [Object],
2025-03-17T02:10:11.6509036Z       [Symbol(shapeMode)]: false,
2025-03-17T02:10:11.6509226Z       [Symbol(kCapture)]: false
2025-03-17T02:10:11.6509381Z     },
2025-03-17T02:10:11.6509524Z     _started: true,
2025-03-17T02:10:11.6509779Z     href: 'https://xx.endpoints.xxxx.cloud.goog/api/v1/project/fffffffff-ffff-ffff-ffff-ffffffffffff',
2025-03-17T02:10:11.6510014Z     req: <ref *3> ClientRequest {
2025-03-17T02:10:11.6510206Z       _events: [Object: null prototype],
2025-03-17T02:10:11.6510371Z       _eventsCount: 5,
2025-03-17T02:10:11.6510556Z       _maxListeners: undefined,
2025-03-17T02:10:11.6510733Z       outputData: [],
2025-03-17T02:10:11.6510902Z       outputSize: 0,
2025-03-17T02:10:11.6511054Z       writable: true,
2025-03-17T02:10:11.6511215Z       destroyed: true,
2025-03-17T02:10:11.6511380Z       _last: false,
2025-03-17T02:10:11.6511537Z       chunkedEncoding: false,
2025-03-17T02:10:11.6511714Z       shouldKeepAlive: true,
2025-03-17T02:10:11.6511913Z       maxRequestsOnConnectionReached: false,
2025-03-17T02:10:11.6512105Z       _defaultKeepAlive: true,
2025-03-17T02:10:11.6512277Z       useChunkedEncodingByDefault: true,
2025-03-17T02:10:11.6512477Z       sendDate: false,
2025-03-17T02:10:11.6512688Z       _removedConnection: false,
2025-03-17T02:10:11.6512858Z       _removedContLen: false,
2025-03-17T02:10:11.6513042Z       _removedTE: false,
2025-03-17T02:10:11.6513222Z       strictContentLength: false,
2025-03-17T02:10:11.6513397Z       _contentLength: 28,
2025-03-17T02:10:11.6513554Z       _hasBody: true,
2025-03-17T02:10:11.6513729Z       _trailer: '',
2025-03-17T02:10:11.6513890Z       finished: true,
2025-03-17T02:10:11.6514120Z       _headerSent: true,
2025-03-17T02:10:11.6514274Z       _closed: false,
2025-03-17T02:10:11.6514447Z       socket: [TLSSocket],
2025-03-17T02:10:11.6514674Z       _header: 'PATCH /api/v1/project/fffffffff-ffff-ffff-ffff-ffffffffffff HTTP/1.1\r\n' +
2025-03-17T02:10:11.6515099Z Finished task execution with error - Error: Failed to update project: 304 - Not Modified
2025-03-17T02:10:11.6515429Z     at DtrackManager._callee3$ (D:\a\_tasks\upload-bom-dtrack_151c89e2-4c73-4d0f-b81b-bbbd418542ba\1.6.0\dtrackManager.js:89:21)
2025-03-17T02:10:11.6515829Z     at tryCatch (D:\a\_tasks\upload-bom-dtrack_151c89e2-4c73-4d0f-b81b-bbbd418542ba\1.6.0\node_modules\@babel\runtime\helpers\regeneratorRuntime.js:45:16)
2025-03-17T02:10:11.6516275Z     at Generator.<anonymous> (D:\a\_tasks\upload-bom-dtrack_151c89e2-4c73-4d0f-b81b-bbbd418542ba\1.6.0\node_modules\@babel\runtime\helpers\regeneratorRuntime.js:133:17)
2025-03-17T02:10:11.6516734Z     at Generator.throw (D:\a\_tasks\upload-bom-dtrack_151c89e2-4c73-4d0f-b81b-bbbd418542ba\1.6.0\node_modules\@babel\runtime\helpers\regeneratorRuntime.js:74:21)
2025-03-17T02:10:11.6517202Z     at asyncGeneratorStep (D:\a\_tasks\upload-bom-dtrack_151c89e2-4c73-4d0f-b81b-bbbd418542ba\1.6.0\node_modules\@babel\runtime\helpers\asyncToGenerator.js:3:17)
2025-03-17T02:10:11.6517632Z     at _throw (D:\a\_tasks\upload-bom-dtrack_151c89e2-4c73-4d0f-b81b-bbbd418542ba\1.6.0\node_modules\@babel\runtime\helpers\asyncToGenerator.js:20:9)
2025-03-17T02:10:11.6517983Z     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2025-03-17T02:10:11.6518303Z         'X-API-Key: odt_ffffffffffffffffffffffffffffffff\r\n' +
2025-03-17T02:10:11.6518525Z         'host: xx.endpoints.xxxx.cloud.goog\r\n' +
2025-03-17T02:10:11.6518729Z         'accept: application/json\r\n' +
2025-03-17T02:10:11.6518915Z         'content-type: application/json\r\n' +
2025-03-17T02:10:11.6519107Z         'content-length: 28\r\n' +
2025-03-17T02:10:11.6519299Z         'Connection: keep-alive\r\n' +
2025-03-17T02:10:11.6519481Z         '\r\n',
2025-03-17T02:10:11.6519637Z       _keepAliveTimeout: 0,
2025-03-17T02:10:11.6519832Z       _onPendingData: [Function: nop],
2025-03-17T02:10:11.6520015Z       agent: [Agent],
2025-03-17T02:10:11.6520172Z       socketPath: undefined,
2025-03-17T02:10:11.6520355Z       method: 'PATCH',
2025-03-17T02:10:11.6520534Z       maxHeaderSize: undefined,
2025-03-17T02:10:11.6520720Z       insecureHTTPParser: undefined,
2025-03-17T02:10:11.6520899Z       joinDuplicateHeaders: undefined,
2025-03-17T02:10:11.6521120Z       path: '/api/v1/project/fffffffff-ffff-ffff-ffff-ffffffffffff',
2025-03-17T02:10:11.6521327Z       _ended: true,
2025-03-17T02:10:11.6521490Z       res: [Circular *2],
2025-03-17T02:10:11.6521648Z       aborted: false,
2025-03-17T02:10:11.6521814Z       timeoutCb: null,
2025-03-17T02:10:11.6521983Z       upgradeOrConnect: false,
2025-03-17T02:10:11.6522139Z       parser: null,
2025-03-17T02:10:11.6522319Z       maxHeadersCount: null,
2025-03-17T02:10:11.6522493Z       reusedSocket: true,
2025-03-17T02:10:11.6522687Z       host: 'xx.endpoints.xxxx.cloud.goog',
2025-03-17T02:10:11.6522871Z       protocol: 'https:',
2025-03-17T02:10:11.6523049Z       [Symbol(shapeMode)]: false,
2025-03-17T02:10:11.6523234Z       [Symbol(kCapture)]: false,
2025-03-17T02:10:11.6523401Z       [Symbol(kBytesWritten)]: 0,
2025-03-17T02:10:11.6523577Z       [Symbol(kNeedDrain)]: false,
2025-03-17T02:10:11.6523749Z       [Symbol(corked)]: 0,
2025-03-17T02:10:11.6523942Z       [Symbol(kOutHeaders)]: [Object: null prototype],
2025-03-17T02:10:11.6524127Z       [Symbol(errored)]: null,
2025-03-17T02:10:11.6524313Z       [Symbol(kHighWaterMark)]: 16384,
2025-03-17T02:10:11.6524522Z       [Symbol(kRejectNonStandardBodyWrites)]: false,
2025-03-17T02:10:11.6524727Z       [Symbol(kUniqueHeaders)]: null
2025-03-17T02:10:11.6524875Z     },
2025-03-17T02:10:11.6525033Z     ntick: true,
2025-03-17T02:10:11.6525210Z     response: [Circular *2],
2025-03-17T02:10:11.6525459Z     originalHost: 'xx.endpoints.xxxx.cloud.goog',
2025-03-17T02:10:11.6525667Z     originalHostHeaderName: 'host',
2025-03-17T02:10:11.6525859Z     responseContent: [Circular *2],
2025-03-17T02:10:11.6526026Z     _ended: true,
2025-03-17T02:10:11.6526183Z     _callbackCalled: true,
2025-03-17T02:10:11.6526358Z     [Symbol(shapeMode)]: false,
2025-03-17T02:10:11.6526535Z     [Symbol(kCapture)]: false
2025-03-17T02:10:11.6526687Z   },
2025-03-17T02:10:11.6526842Z   toJSON: [Function: responseToJSON],
2025-03-17T02:10:11.6527030Z   caseless: Caseless {
2025-03-17T02:10:11.6527190Z     dict: {
2025-03-17T02:10:11.6527362Z       date: 'Mon, 17 Mar 2025 02:10:11 GMT',
2025-03-17T02:10:11.6527571Z       'x-powered-by': 'Dependency-Track v4.12.6',
2025-03-17T02:10:11.6527792Z       'cache-control': 'private, max-age=0, must-revalidate, no-cache',
2025-03-17T02:10:11.6528024Z       'access-control-allow-origin': '*',
2025-03-17T02:10:11.6528234Z       'access-control-allow-methods': 'GET, POST, PUT, DELETE, OPTIONS',
2025-03-17T02:10:11.6528552Z       'access-control-allow-headers': 'Origin, Content-Type, Authorization, X-Requested-With, Content-Length, Accept, Origin, X-Api-Key, X-Total-Count, *',
2025-03-17T02:10:11.6528920Z       'access-control-expose-headers': 'Origin, Content-Type, Authorization, X-Requested-With, Content-Length, Accept, Origin, X-Api-Key, X-Total-Count',
2025-03-17T02:10:11.6529211Z       'access-control-allow-credentials': 'true',
2025-03-17T02:10:11.6529452Z       'access-control-max-age': '3600',
2025-03-17T02:10:11.6529670Z       'x-request-id': '35575628-af2c-4204-8f72-b6bc2f738d6e',
2025-03-17T02:10:11.6530082Z       'content-security-policy': "default-src 'self';script-src 'self' 'unsafe-inline';style-src 'self' 'unsafe-inline';img-src 'self';connect-src 'self';font-src 'self';object-src 'self';media-src 'self';frame-src 'self';child-src 'self';form-action 'self';frame-ancestors 'none';",
2025-03-17T02:10:11.6530476Z       'x-frame-options': 'DENY',
2025-03-17T02:10:11.6530659Z       'content-length': '0',
2025-03-17T02:10:11.6530819Z       via: '1.1 google',
2025-03-17T02:10:11.6531039Z       'alt-svc': 'h3=":443"; ma=2592000,h3-29=":443"; ma=2592000'
2025-03-17T02:10:11.6531231Z     }
2025-03-17T02:10:11.6531361Z   },
2025-03-17T02:10:11.6531526Z   [Symbol(shapeMode)]: true,
2025-03-17T02:10:11.6531700Z   [Symbol(kCapture)]: false,
2025-03-17T02:10:11.6531875Z   [Symbol(kHeaders)]: {
2025-03-17T02:10:11.6532052Z     date: 'Mon, 17 Mar 2025 02:10:11 GMT',
2025-03-17T02:10:11.6532271Z     'x-powered-by': 'Dependency-Track v4.12.6',
2025-03-17T02:10:11.6532495Z     'cache-control': 'private, max-age=0, must-revalidate, no-cache',
2025-03-17T02:10:11.6532712Z     'access-control-allow-origin': '*',
2025-03-17T02:10:11.6532919Z     'access-control-allow-methods': 'GET, POST, PUT, DELETE, OPTIONS',
2025-03-17T02:10:11.6533224Z     'access-control-allow-headers': 'Origin, Content-Type, Authorization, X-Requested-With, Content-Length, Accept, Origin, X-Api-Key, X-Total-Count, *',
2025-03-17T02:10:11.6533603Z     'access-control-expose-headers': 'Origin, Content-Type, Authorization, X-Requested-With, Content-Length, Accept, Origin, X-Api-Key, X-Total-Count',
2025-03-17T02:10:11.6533894Z     'access-control-allow-credentials': 'true',
2025-03-17T02:10:11.6534084Z     'access-control-max-age': '3600',
2025-03-17T02:10:11.6534298Z     'x-request-id': '35575628-af2c-4204-8f72-b6bc2f738d6e',
2025-03-17T02:10:11.6534710Z     'content-security-policy': "default-src 'self';script-src 'self' 'unsafe-inline';style-src 'self' 'unsafe-inline';img-src 'self';connect-src 'self';font-src 'self';object-src 'self';media-src 'self';frame-src 'self';child-src 'self';form-action 'self';frame-ancestors 'none';",
2025-03-17T02:10:11.6535076Z     'x-frame-options': 'DENY',
2025-03-17T02:10:11.6535258Z     'content-length': '0',
2025-03-17T02:10:11.6535418Z     via: '1.1 google',
2025-03-17T02:10:11.6535624Z     'alt-svc': 'h3=":443"; ma=2592000,h3-29=":443"; ma=2592000'
2025-03-17T02:10:11.6535879Z   },
2025-03-17T02:10:11.6536031Z   [Symbol(kHeadersCount)]: 30,
2025-03-17T02:10:11.6536213Z   [Symbol(kTrailers)]: null,
2025-03-17T02:10:11.6536394Z   [Symbol(kTrailersCount)]: 0
2025-03-17T02:10:11.6536547Z }
2025-03-17T02:10:11.6537464Z ##[error]Failed to update project: 304 - Not Modified
2025-03-17T02:10:11.7017118Z ##[error]Exit code 1 returned from process: file name 'C:\agents\4.252.0\externals\node20_1\bin\node.exe', arguments '"D:\a\_tasks\upload-bom-dtrack_151c89e2-4c73-4d0f-b81b-bbbd418542ba\1.6.0\task.js"'.
2025-03-17T02:10:11.7022653Z ##[section]Finishing: Upload BOM to https://xx.endpoints.xxxx.cloud.goog

Thanks.

--- a/UploadBOM/src/dtrackClient.js
+++ b/UploadBOM/src/dtrackClient.js
@@ -158,6 +158,8 @@ class DTrackClient {
         }
       });
 
+      console.log(data);
+
       request(`/api/v1/project/${projId}`, {
         ...this.baseOptions,
         method: 'PATCH',
@@ -166,7 +168,6 @@ class DTrackClient {
         if (!error && response.statusCode === 200) {
           resolve(response.body);
         } else {
-          console.log('Error:', response);
           reject({ error, response });
         }
       });

--- a/UploadBOM/src/dtrackClient.js
+++ b/UploadBOM/src/dtrackClient.js
@@ -148,7 +148,7 @@ class DTrackClient {
         "classifier": classifier,
         "swidTagId": swidTagId,
         "group": group,
-        "tags": tags ? tags.map(tag => ({ name: tag })) : null,
+        "tags": tags,
       }
 
       // Remove properties with null values
@@ -157,8 +157,6 @@ class DTrackClient {
           delete data[key];
         }
       });
-
-      console.log(data);
 
       request(`/api/v1/project/${projId}`, {
         ...this.baseOptions,

--- a/UploadBOM/src/dtrackManager.js
+++ b/UploadBOM/src/dtrackManager.js
@@ -1,4 +1,4 @@
-import {localize} from './localization.js'
+import { localize } from './localization.js'
 import Utils from './utils.js'
 
 class DtrackManager {
@@ -25,28 +25,37 @@ class DtrackManager {
   async updateProject(projectId, description, classifier, swidTagId, group, tags) {
     try {
       let updatedInfo = {};
-      tags = tags ? tags.map(tag => ({ name: tag })) : null
-      
+
       let projectInfo = await this.getProjectInfo(projectId);
       if (projectInfo) {
-        if(description && projectInfo.description !== description) {
+        if (description && projectInfo.description !== description) {
           updatedInfo.description = description;
         }
 
-        if(classifier && projectInfo.classifier !== classifier) {
+        if (classifier && projectInfo.classifier !== classifier) {
           updatedInfo.classifier = classifier;
         }
 
-        if(swidTagId && projectInfo.swidTagId !== swidTagId) {
+        if (swidTagId && projectInfo.swidTagId !== swidTagId) {
           updatedInfo.swidTagId = swidTagId;
         }
 
-        if(group && projectInfo.group !== group) {
+        if (group && projectInfo.group !== group) {
           updatedInfo.group = group;
         }
 
-        if(tags && projectInfo.tags !== tags) {
-          updatedInfo.tags = tags;
+        if (tags) {
+          // Get existing and new tags normalized for comparison
+          const existingTagNames = projectInfo.tags?.map(tag => tag.name.toLowerCase()).sort() || [];
+          const newTagNames = [...tags].map(tag => tag.toLowerCase()).sort();
+
+          // Check if arrays are different either in length or content
+          const tagsAreDifferent = existingTagNames.length !== newTagNames.length ||
+            existingTagNames.some((tag, i) => tag !== newTagNames[i]);
+
+          if (tagsAreDifferent) {
+            updatedInfo.tags = tags.map(tag => ({ name: tag }));
+          }
         }
       }
 
@@ -59,7 +68,7 @@ class DtrackManager {
       }
 
       console.log(localize('UpdatingProject'));
-      const newSettings = await this.dtrackClient.updateProject(projectId, description, classifier, swidTagId, group, tags);
+      const newSettings = await this.dtrackClient.updateProject(projectId, updatedInfo.description, updatedInfo.classifier, updatedInfo.swidTagId, updatedInfo.group, updatedInfo.tags);
 
       console.log(localize('NewProjectSettings'));
       console.log(localize('projectSettings', projectId, newSettings.name, newSettings.version, newSettings.description, newSettings.classifier, newSettings.swidTagId, newSettings.group, newSettings.tags.map(tag => tag.name).join(', '), newSettings.isLatest));
@@ -117,7 +126,7 @@ class DtrackManager {
   async waitMetricsRefresh(projectId) {
     const lastBomImport = new Date((await this.getProjectInfo(projectId)).lastBomImport);
     let lastOccurrence = undefined;
-      
+
     do {
       await Utils.sleepAsync(2000);
       console.log(localize('Polling'));

--- a/UploadBOM/src/dtrackManager.js
+++ b/UploadBOM/src/dtrackManager.js
@@ -24,7 +24,46 @@ class DtrackManager {
 
   async updateProject(projectId, description, classifier, swidTagId, group, tags) {
     try {
-      await this.dtrackClient.updateProject(projectId, description, classifier, swidTagId, group, tags);
+      let updatedInfo = {};
+      tags = tags ? tags.map(tag => ({ name: tag })) : null
+      
+      let projectInfo = await this.getProjectInfo(projectId);
+      if (projectInfo) {
+        if(description && projectInfo.description !== description) {
+          updatedInfo.description = description;
+        }
+
+        if(classifier && projectInfo.classifier !== classifier) {
+          updatedInfo.classifier = classifier;
+        }
+
+        if(swidTagId && projectInfo.swidTagId !== swidTagId) {
+          updatedInfo.swidTagId = swidTagId;
+        }
+
+        if(group && projectInfo.group !== group) {
+          updatedInfo.group = group;
+        }
+
+        if(tags && projectInfo.tags !== tags) {
+          updatedInfo.tags = tags;
+        }
+      }
+
+      console.log(localize('CurrentProjectSettings'));
+      console.log(localize('projectSettings', projectId, projectInfo.name, projectInfo.version, projectInfo.description, projectInfo.classifier, projectInfo.swidTagId, projectInfo.group, projectInfo.tags.map(tag => tag.name).join(', '), projectInfo.isLatest));
+
+      if (Object.keys(updatedInfo).length === 0) {
+        console.log(localize('NoProjectChanges'));
+        return;
+      }
+
+      console.log(localize('UpdatingProject'));
+      const newSettings = await this.dtrackClient.updateProject(projectId, description, classifier, swidTagId, group, tags);
+
+      console.log(localize('NewProjectSettings'));
+      console.log(localize('projectSettings', projectId, newSettings.name, newSettings.version, newSettings.description, newSettings.classifier, newSettings.swidTagId, newSettings.group, newSettings.tags.map(tag => tag.name).join(', '), newSettings.isLatest));
+
     }
     catch (err) {
       throw new Error(localize('ProjectUpdateFailed', Utils.getErrorMessage(err)));

--- a/UploadBOM/src/localization.js
+++ b/UploadBOM/src/localization.js
@@ -5,6 +5,6 @@ function localize(key, ...params) {
     return loc(key, ...params);
   }
   
-  return loc(key)
+  return loc(key);
 }
 export {localize};

--- a/UploadBOM/src/task.js
+++ b/UploadBOM/src/task.js
@@ -76,7 +76,6 @@ const run = async () => {
   console.log(localize('BOMUploadSucceed', token));
 
   if (shouldUpdateProject(params)) {
-    console.log(localize('UpdatingProject'));
     await dtrackManager.updateProject(projectId, params.projectDescription, params.projectClassifier, params.projectSwidTagId, params.projectGroup, params.projectTags);
   }
 

--- a/UploadBOM/src/task.json
+++ b/UploadBOM/src/task.json
@@ -312,7 +312,11 @@
     "MissingProjectInfoWhenAutoCreate": "Project name and version is required when auto creating project.",
     "MissingProjectInfoWhenNoProjectId": "Project name and version is required when Project Id is not set.",
     "ProjectUpdateFailed": "Failed to update project: %s",
-    "UpdatingProject": "Updating project."
+    "UpdatingProject": "Updating project.",
+    "NoProjectChanges": "Project settings don't need to be updated.",
+    "CurrentProjectSettings": "Current project settings.",
+    "NewProjectSettings": "Update succeeded. New project settings.",
+    "projectSettings": "Id: %s\nName: %s\nVersion: %s\nDescription: %s\nClassifier: %s\nSWID Tag ID: %s\nGroup: %s\nTags: %s\nIs Latest: %s"
   },
   "execution": {
     "Node20_1": {

